projects/components/src/select/select.component.ts

```diff
@@ -89,37 +97,59 @@ import { SelectSize } from './select-size';
               class="trigger-content menu-with-background"
               [ngClass]="[this.justifyClass]"
             >
-              <ht-label class="trigger-label" [label]="selected?.selectedLabel || selected?.label || this.placeholder">
-              </ht-label>
+              <ng-container
+                [ngTemplateOutlet]="selected?.selectOptionRenderer?.getTemplateRef() ?? defaultMenuWithBackgroundTriggerTemplate"
+              ></ng-container>
+              <ng-template #defaultMenuWithBackgroundTriggerTemplate
+                ><<ht-label
+                  class="trigger-label"
+                  [label]="selected?.selectedLabel || selected?.label || this.placeholder"
+                >
+                </ht-label
+              ></ng-template>
               <ht-icon class="trigger-icon" icon="${IconType.ChevronDown}" size="${IconSize.Small}"> </ht-icon>
             </div>
           </div>
         </ht-popover-trigger>
         <ht-popover-content>
           <div class="select-content" [ngStyle]="{ 'minWidth.px': triggerContainer.offsetWidth }">
+            <!-- Top Control Item --->
             <ng-container *htLetAsync="this.topControlItems$ as topControlItems">
               <div *ngIf="topControlItems?.length !== 0">
-                <ng-container
-                  *ngTemplateOutlet="itemsTemplate; context: { items: topControlItems, showSelectionStatus: false }"
-                ></ng-container>
-
+                <div
+                  *ngFor="let item of topControlItems"
+                  (click)="this.onSelectionChange(item)"
+                  class="select-option"
+                  [ngClass]="this.getStyleClassesForSelectItem | htMemoize: this.size:item"
+                >
+                  <ng-container
+                    *ngTemplateOutlet="item.selectOptionRenderer?.getTemplateRef() ?? defaultSelectOptionTemplate; context: {$implicit: item}"
+                  ></ng-container>
+                </div>
                 <ht-divider></ht-divider>
               </div>
             </ng-container>
 
-            <ng-container
-              *ngTemplateOutlet="itemsTemplate; context: { items: items, showSelectionStatus: true }"
-            ></ng-container>
-          </div>
-
-          <ng-template #itemsTemplate let-items="items" let-showSelectionStatus="showSelectionStatus">
+            <!-- Select Option Items --->
             <div
-              *ngFor="let item of items"
+              *ngFor="let item of this.items"
```

Comment:
probably could have left the items template and just pulled the item template out to avoid the repeated ngFors, but seems fine functionally.

Reply:
Updated